shared/osarch.go

```diff
@@ -0,0 +1,54 @@
+package shared
+
+import (
+	"fmt"
+
+	"github.com/lxc/lxd/shared/osarch"
+)
+
+var alpineLinuxArchitectureNames = map[int]string{
+	osarch.ARCH_32BIT_INTEL_X86:           "x86",
+	osarch.ARCH_32BIT_ARMV7_LITTLE_ENDIAN: "armhf",
+}
+
+var centosArchitectureNames = map[int]string{
+	osarch.ARCH_32BIT_INTEL_X86: "i386",
+}
+
+var debianArchitectureNames = map[int]string{
+	osarch.ARCH_32BIT_INTEL_X86:             "i386",
+	osarch.ARCH_64BIT_INTEL_X86:             "amd64",
+	osarch.ARCH_32BIT_ARMV7_LITTLE_ENDIAN:   "armhf",
+	osarch.ARCH_64BIT_ARMV8_LITTLE_ENDIAN:   "arm64",
+	osarch.ARCH_32BIT_POWERPC_BIG_ENDIAN:    "powerpc",
+	osarch.ARCH_64BIT_POWERPC_BIG_ENDIAN:    "powerpc64",
+	osarch.ARCH_64BIT_POWERPC_LITTLE_ENDIAN: "ppc64el",
+}
+
+var distroArchitecture = map[string]map[int]string{
+	"alpinelinux": alpineLinuxArchitectureNames,
+	"centos":      centosArchitectureNames,
+	"debian":      debianArchitectureNames,
+	"ubuntu":      debianArchitectureNames,
```

Comment:
Not sure that this particular entry makes sense. I'd be perfectly fine with the Ubuntu yaml setting an architecture map of "debian" as that's the architecture naming scheme it's using.